test/ref_ops_test.cpp

```diff
@@ -4257,6 +4260,195 @@ TEST_CASE(scatter_test)
     }
 }
 
+TEST_CASE(scatter_reduction_test)
+{
+    {
+        // Test sub-ops for the three reduction values scatter_none, scatter_add, scatter_mul
+
+        migraphx::program p;
+        auto* mm = p.get_main_module();
+        migraphx::shape sd{migraphx::shape::float_type, {1, 5}};
+        std::vector<float> vd({1., 2., 3., 4., 5.});
+
+        migraphx::shape si{migraphx::shape::int32_type, {1, 2}};
+        std::vector<int> vi = {1, 3};
+
+        migraphx::shape su{migraphx::shape::float_type, {1, 2}};
+        std::vector<float> vu = {1.1, 2.1};
+
+        auto ld = mm->add_literal(migraphx::literal{sd, vd});
+        auto li = mm->add_literal(migraphx::literal{si, vi});
+        auto lu = mm->add_literal(migraphx::literal{su, vu});
+        auto r  = mm->add_instruction(migraphx::make_op("scatter_none", {{"axis", 1}}), ld, li, lu);
+        mm->add_return({r});
+        p.compile(migraphx::ref::target{});
+        auto result = p.eval({}).back();
+        std::vector<float> results_vector;
+        result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+        std::vector<float> gold_none = {1.0, 1.1, 3.0, 2.1, 5.0};
+        EXPECT(migraphx::verify_range(results_vector, gold_none));
+    }
+
+    {
+        migraphx::program p;
+        auto* mm = p.get_main_module();
+        migraphx::shape sd{migraphx::shape::float_type, {1, 5}};
+        std::vector<float> vd({1., 2., 3., 4., 5.});
+
+        migraphx::shape si{migraphx::shape::int32_type, {1, 2}};
+        std::vector<int> vi = {1, 3};
+
+        migraphx::shape su{migraphx::shape::float_type, {1, 2}};
+        std::vector<float> vu = {1.1, 2.1};
+
+        auto ld = mm->add_literal(migraphx::literal{sd, vd});
+        auto li = mm->add_literal(migraphx::literal{si, vi});
+        auto lu = mm->add_literal(migraphx::literal{su, vu});
+        auto r  = mm->add_instruction(migraphx::make_op("scatter_mul", {{"axis", 1}}), ld, li, lu);
+        mm->add_return({r});
+        p.compile(migraphx::ref::target{});
+        auto result = p.eval({}).back();
+        std::vector<float> results_vector;
+        result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+        std::vector<float> gold_mul = {1.0, 2.2, 3.0, 8.4, 5.0};
+
+        EXPECT(migraphx::verify_range(results_vector, gold_mul));
+    }
+
+    {
+        migraphx::program p;
```

Comment:
As @pfultz2 mentioned, you can separate these into different unit tests. Also, you can make a common function above the first three tests that constructs a program and takes in the scatter mode as a parameter:

```
migraphx::program create_scatter_program(std::string scatter_mode) {
// ... add instructions and literals here
}
```

Reply:
changed